client/app/domain/domain-tabs.controller.js

```diff
@@ -130,14 +132,22 @@ angular.module('App').controller(
       this.$location.search('tab', this.selectedTab);
     }
 
-    static getUpdateOwnerUrl(domain) {
-      if (
-        _.get(domain, 'whoisOwner', false)
-        && domain.whoisOwner !== 'pending'
-      ) {
-        return `#/useraccount/contact/${domain.name}/${domain.whoisOwner.id}`;
+    getUpdateOwnerUrl(domain) {
+      const ownerUrlInfo = { target: '', error: '' };
+      if (_.has(domain, 'name') && _.has(domain, 'whoisOwner.id')) {
+        ownerUrlInfo.target = `#/useraccount/contact/${domain.name}/${domain.whoisOwner.id}`;
+      } else {
+        ownerUrlInfo.error = this.$scope.tr('domain_dashboard_whois_error', [
+          (!_.has(domain, 'name')) ? this.$scope.tr('domain_tab_REDIRECTION_add_step4_server_cname_error') : domain.whoisOwner,
+        ]);
+      }
+      return ownerUrlInfo;
+    }
+
+    handleOwnerUrlError(errMsg) {
+      if (errMsg) {
```

Comment:
You should throw an error when a parameter is incorrect

Reply:
We are already handling the error (if present), by showing an alerter. Do you think we will have to throw it once again?